### PR TITLE
fix(scene composer): Color picker bug fixes

### PR DESCRIPTION
**Pull request by divya-sea**
* bug fixes for color picker
## Overview
Provide an explanation of what the change is

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx [hunk 1]
```diff
@@ -69,22 +86,62 @@ export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps):
...
+      setHexCodeError(''); // Clear any existing error message
+      setNewColor(color.hex);
+      onSelectColor(color.hex);
+    },
+    [color, onSelectColor, onUpdateCustomColors],
+  );
+
+  const handleHexCodeChange = useCallback(
+    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
+      setNewColor(event.detail.value);
```
Comment (haweston) [inline packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:105]: Do we want to set the color if it's not valid hex code?

File: packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx [hunk 2]
```diff
@@ -34,30 +41,40 @@ export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps):
...
-  }, []);
+
+    return () => {
+      document.removeEventListener('click', handleOutsideClick);
+    };
+  }, [showPicker, handleOutsideClick]);
+
+  const checkIfCustomColor = (color: any) => {
+    if (!Object.values(palleteColors).includes(color)) {
+      setCustomInternalColors(customInternalColors.concat(color));
```
Comment (haweston) [inline packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:70]: What's the experience here? Are we tracking all previous attempted custom colors?

Comment (divya-sea) [reply packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:70]: Experience is we provide basic colors that will be displayed by default to all. But of they chose from the StechPicker and that hex code is part of the basic default color then we don't want to add those colors to part of the custom colors. This requirement is from UX. So this condition will prevent adding redundant colors to custom color list.

File: packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx (hunk 1 above)
Comment (divya-sea) [reply packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:105]: First, we want to set it to the state variable as this is coming as external prop. Then we are validating whether it is a valid color or not. In the same method we are validating.

Comment (haweston) [reply packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:105]: in handleHexCodeChange it doesn't look like an external prop yet. it's an internal event.detail.value that we haven't validated yet?

Comment (divya-sea) [reply packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx:105]: const [newColor, setNewColor] = useState<string>(color);
Color is a prop and I am using a single state variable to update the state of the component. So initially will have a value that is coming from prop. If user change the hex code then we are checking whther its a valid or not then resetting it again.
const handleHexCodeChange = useCallback(
    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
      setNewColor(event.detail.value);
}

**Review by haweston — APPROVED** [review]

**Pull request merged**